vcfconcat.c

```diff
@@ -549,6 +553,114 @@ static void concat(args_t *args)
     }
 }
 
+BGZF *hts_get_bgzfp(htsFile *fp);
+static void naive_concat(args_t *args)
+{
+    // only compressed BCF atm
+
+    BGZF *bgzf_out;
+    if ( strcmp("-",args->output_fname) )
+        bgzf_out = bgzf_open(args->output_fname,"w");
+    else
+        bgzf_out = bgzf_dopen(fileno(stdout), "w");
+
+    int page_size = getpagesize();
+    char *buf = (char*) valloc(page_size);
```

Comment:
I had no doubts you'd come to justify anything horrid, even hard-wired machine-specific constants, as long as it was you who put it there :-D
It is true that last time I did the benchmarking of this was ~10 years ago (not in 1983), but it did make a significant difference. It is possible that code optimization, plenty of available memory and better machines make this practice obsolete, but 1) can you claim you have benchmarked this on every machine out there 2) what is the harm?